recipes/slurm_install.rb

```diff
@@ -16,43 +16,64 @@
 include_recipe 'aws-parallelcluster::base_install'
 include_recipe 'aws-parallelcluster::munge_install'
 
-slurm_tarball = "#{node['cfncluster']['sources_dir']}/slurm-#{node['cfncluster']['slurm']['version']}.tar.gz"
+case node['cfncluster']['cfn_node_type']
+when 'MasterServer'
+  slurm_tarball = "#{node['cfncluster']['sources_dir']}/slurm-#{node['cfncluster']['slurm']['version']}.tar.gz"
 
-# Get slurm tarball
-remote_file slurm_tarball do
-  source node['cfncluster']['slurm']['url']
-  mode '0644'
-  retries 3
-  retry_delay 5
-  # TODO: Add version or checksum checks
-  not_if { ::File.exist?(slurm_tarball) }
-end
+  # Get slurm tarball
+  remote_file slurm_tarball do
+    source node['cfncluster']['slurm']['url']
+    mode '0644'
+    retries 3
+    retry_delay 5
+    # TODO: Add version or checksum checks
+    not_if { ::File.exist?(slurm_tarball) }
+  end
 
-# Install Slurm
-bash 'make install' do
-  user 'root'
-  group 'root'
-  cwd Chef::Config[:file_cache_path]
-  code <<-SLURM
-    tar xf #{slurm_tarball}
-    cd slurm-slurm-#{node['cfncluster']['slurm']['version']}
-    ./configure --prefix=/opt/slurm
-    CORES=$(grep processor /proc/cpuinfo | wc -l)
-    make -j $CORES
-    make install
-    make install-contrib
-  SLURM
-  # TODO: Fix, so it works for upgrade
-  creates '/opt/slurm/bin/srun'
-end
+  # Install Slurm
+  bash 'make install' do
+    user 'root'
+    group 'root'
+    cwd Chef::Config[:file_cache_path]
+    code <<-SLURM
+      tar xf #{slurm_tarball}
+      cd slurm-slurm-#{node['cfncluster']['slurm']['version']}
+      ./configure --prefix=/opt/slurm
+      CORES=$(grep processor /proc/cpuinfo | wc -l)
+      make -j $CORES
+      make install
+      make install-contrib
+    SLURM
+    # TODO: Fix, so it works for upgrade
+    creates '/opt/slurm/bin/srun'
+  end
+
+  # Setup slurm user
+  user "slurm" do
```

Comment:
The user must be added in the compute node too.

Reply:
done